clientcontroller/babylon.go

```diff
@@ -174,16 +182,90 @@ func (bc *BabylonController) SubmitCovenantSigs(covSigs []*types.CovenantSigs) (
 			SlashingUnbondingTxSigs: covSig.SlashingUnbondingSigs,
 		})
 	}
-	res, err := bc.reliablySendMsgs(msgs)
-	if err != nil {
-		return nil, err
+
+	return bc.reliablySendMsgsResendingOnMsgErr(msgs)
+}
+
+// reliablySendMsgsResendingOnMsgErr sends the msgs to the chain, if some msg fails to execute
+// and contains 'message index: %d', it will remove that msg from the batch and send again
+// if there is no more message available, returns the last error.
+func (bc *BabylonController) reliablySendMsgsResendingOnMsgErr(msgs []sdk.Msg) (*types.TxResponse, error) {
```

Comment:
I must say this whole functions makes me a bit nervious without e2e test. Do you think it would be easy to setup e2e test to only test 
`func (bc *BabylonController) SubmitCovenantSigs(covSigs []*types.CovenantSigs) (*types.TxResponse, error) {`
function, and not whole covenant logic ? 

In this test:
- create babylon node with 1 covenat committe member configures
- create 3 delegations with this covenant member in script
- then send batch with invlaid sigs for 2 of them, and valid for the last one
the send should succeed for the last delegation

This can be done in separate PR.

Reply:
Well, if we are not certain about this function, I believe the e2e test should be done prior to merge 

I will work on this